server.go

```diff
@@ -469,6 +474,15 @@ func (s *Server) Serve(lis net.Listener) error {
 			s.mu.Lock()
 			s.printf("done serving; Accept = %v", err)
 			s.mu.Unlock()
+
+			// If Stop or GracefulStop is called, block until they are done and return nil
+			select {
+			case <-s.quit:
```

Comment:
A server can actually `Serve` on multiple listeners. This only works in the case of one `Serve`.

Reply:
ah good catch, will fix!

Reply:
@menghanl Fixed by using a `sync.Once` to close the channels, this way when `s.quit` or `s.done` is closed, all go-routines running `Serve()` will be unblocked.